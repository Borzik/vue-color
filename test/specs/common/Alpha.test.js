import { shallow } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'
import randomInt from 'random-int'

import Alpha from '../../../src/components/common/Alpha'

describe('Alpha.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(Alpha)
  })

  test('renders correctly', () => {
    const renderer = createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })

  test('click', () => {
    wrapper.vm.$refs.container = { clientWidth: 100, getBoundingClientRect () { return { left: 0 } } }
    wrapper.vm.handleChange({pageX: 60, preventDefault () {}})
    expect(wrapper.vm.$data._color.a).toBe(0.6)
  })

  test('should not exceed min and max', () => {
    wrapper.setProps({
      color: {
        h: randomInt(360),
        s: randomInt(100),
        l: randomInt(100),
        a: 0
      }
    })

    const pointer = wrapper.find('.vc-alpha-pointer')
    expect(pointer.element.style.left).toBe('0%')

    wrapper.setProps({
      color: {
        h: randomInt(360),
        s: randomInt(100),
        l: randomInt(100),
        a: 1
      }
    })

    expect(pointer.hasStyle('left', '100%')).toBe(true)

    wrapper.vm.handleChange({pageX: -1, preventDefault () {}})
    expect(wrapper.vm.$data._color.a).toBe(0)

    wrapper.vm.$refs.container = { clientWidth: 100, getBoundingClientRect () { return { left: 0 } } }
    wrapper.vm.handleChange({pageX: 101, preventDefault () {}})
    expect(wrapper.vm.$data._color.a).toBe(1)
    expect(pointer.element.style.left).toBe('100%')
  })
})
