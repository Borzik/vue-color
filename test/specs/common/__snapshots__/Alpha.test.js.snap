// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alpha.vue renders correctly 1`] = `
<div class="vc-alpha">
    <div class="vc-alpha-checkboard-wrap">
        <!---->
    </div>
    <div class="vc-alpha-gradient" style="background:linear-gradient(to right, rgba(25,77,51, 0) 0%, rgba(25,77,51, 1) 100%);"></div>
    <div class="vc-alpha-container">
        <div class="vc-alpha-pointer" style="left:100%;">
            <div class="vc-alpha-picker"></div>
        </div>
    </div>
</div>
`;
